### ISSUE TITLE
Trying to connect to Vagrant machine with Remote SSH

### ISSUE THREAD

**Issue by monoprosito** [microsoft/vscode-remote-release#2848]
I already tried this on VSCode Insiders and I'm getting it wrong too.

- VSCode Version: 1.44.2
- Local OS Version: Windows 10 Version 1909 (OS Build 18363.778)
- Remote OS Version: Ubuntu 14.04
- Remote Extension/Connection Type: SSH
- ms-vscode-remote.remote-ssh Version: 0.51.0
- ms-vscode-remote.remote-ssh-edit Version: 0.51.0

![...](https://i.imgur.com/KXITjc4.png)

VSCode Output:
```[13:05:27.368] Log Level: 2
[13:05:27.374] remote-ssh@0.51.0
[13:05:27.374] win32 x64
[13:05:27.380] SSH Resolver called for "ssh-remote+default", attempt 1
[13:05:27.381] SSH Resolver called for host: default
[13:05:27.381] Setting up SSH remote "default"
[13:05:27.437] Using commit id "ff915844119ce9485abfe8aa9076ec76b5300ddd" and quality "stable" for server
[13:05:27.440] Install and start server if needed
[13:05:29.745] Checking ssh with "ssh -V"
[13:05:29.826] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[13:05:29.831] Running script with connection command: ssh -T -D 58894 default bash
[13:05:29.842] Terminal shell path: C:\Windows\System32\cmd.exe
[13:05:29.989] > 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> ]0;C:\Windows\System32\cmd.exe
[13:05:29.991] Got some output, clearing connection timeout
[13:05:29.996] > 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
[13:05:30.069] > The authenticity of host '[127.0.0.1]:2222 ([127.0.0.1]:2222)' can't be establis
> hed.
> ECDSA key fingerprint is SHA256:accSIBZ+3ujYjOf3QGPotv/Npt4kzkUWklaxDUu9+0E.    
> Are you sure you want to continue connecting (yes/no)?
[13:05:30.070] Detected fingerprint confirmation message
[13:05:30.071] Showing fingerprint confirmation dialog
[13:05:49.154] Got fingerprint response: yes
[13:05:49.156] "install" wrote data to terminal: "yes"
[13:05:49.180] > y
[13:05:49.189] > es
> 
[13:05:49.241] > Warning: Permanently added '[127.0.0.1]:2222' (ECDSA) to the list of known hosts
> .
> 
[13:05:50.763] > 9ff965b44472: running
> 
[13:05:50.842] > Acquiring lock on /home/vagrant/.vscode-server/bin/ff915844119ce9485abfe8aa9076e
> c76b5300ddd/vscode-remote-lock.vagrant.ff915844119ce9485abfe8aa9076ec76b5300ddd 
> 
[13:05:50.852] > \ln /home/vagrant/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/vs
> code-remote-lock.vagrant.ff915844119ce9485abfe8aa9076ec76b5300ddd.target /home/v
> agrant/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/vscode-remote
> -lock.vagrant.ff915844119ce9485abfe8aa9076ec76b5300ddd
> 
[13:05:50.869] > Installing to /home/vagrant/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b
> 5300ddd...
> 
[13:05:50.877] > 9ff965b44472%%1%%
> 
[13:05:50.893] > Downloading with wget
> 
[13:06:01.499] > Download complete
> 
[13:06:01.508] > 9ff965b44472%%2%%
> 
[13:06:02.380] > tar: 
[13:06:02.389] > Skipping to next header
> 
[13:06:03.011] > tar: 
[13:06:03.020] > tar: Exiting with failure status due to previous errors
> WARNING: tar exited with non-0 exit code
> 
[13:06:03.029] > WARNING: /home/vagrant/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300d
> dd/server.sh doesn't exist. Download/untar may have failed.
> 
[13:06:03.080] > XDG_SESSION_ID=4
> SHELL=/bin/bash
> SSH_CLIENT=10.0.2.2 58896 22
> USER=vagrant
> VSCODE_AGENT_FOLDER=/home/vagrant/.vscode-server
> PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/us
> r/local/games
> MAIL=/var/mail/vagrant
> PWD=/home/vagrant
> LANG=en_US.UTF-8
> HOME=/home/vagrant
> SHLVL=2
> LOGNAME=vagrant
[13:06:03.086] > 
> SSH_CONNECTION=10.0.2.2 58896 10.0.2.15 22
> XDG_RUNTIME_DIR=/run/user/1000
> _=/usr/bin/printenv
> OLDPWD=/home/vagrant/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd
[13:06:03.096] > 
> 
> Starting server with command... /home/vagrant/.vscode-server/bin/ff915844119ce94
> 85abfe8aa9076ec76b5300ddd/server.sh --host=127.0.0.1 --enable-remote-auto-shutdo
> wn  --port=0 &> "/home/vagrant/.vscode-server/.ff915844119ce9485abfe8aa9076ec76b
> 5300ddd.log" < /dev/null
[13:06:03.107] > 
> Waiting for server log...
> 
[13:06:03.619] > Waiting for server log...
> 
[13:06:04.139] > Waiting for server log...
> 
[13:06:04.659] > Waiting for server log...
> 
[13:06:05.178] > Waiting for server log...
> 
[13:06:05.706] > Waiting for server log...
> 
[13:06:06.220] > Waiting for server log...
> 
[13:06:06.730] >  
> *
[13:06:06.737] > 
> * Reminder: You may only use this software with Visual Studio family products,  
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057) 
> *
> 
[13:06:06.744] > 
> 
[13:06:06.764] > Server did not start successfully. Full server log at /home/vagrant/.vscode-serv
> er/.ff915844119ce9485abfe8aa9076ec76b5300ddd.log >>>
> bash: line 371: /home/vagrant/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec7
> 6b5300ddd/server.sh: No such file or directory
[13:06:06.775] > 
> <<< End of server log
> 9ff965b44472##32##
[13:06:06.776] Received install output: 9ff965b44472##32##
[13:06:06.777] Resolver error: The VS Code Server failed to start
[13:06:06.787] ------




[13:06:07.373] > 
> 
[13:06:07.664] "install" terminal command done
[13:06:07.667] Install terminal quit with output:
```

Steps to Reproduce:

1. My `%USERPROFILE%/.ssh/config`

```
# Read more about SSH config files: https://linux.die.net/man/5/ssh_config
Host default
    User vagrant
    HostName 127.0.0.1
    Port 2222
    IdentityFile "C:/Users/A Monkey Programmer/Holberton/u1404/.vagrant/machines/default/virtualbox/private_key"
```

2. After having my SSH configuration, I click on the action **Connect to Host in Current Window** and when selecting the remote host platform **Linux**, the final result is the error log I put above

NOTE: I've done a lot of research on this topic and haven't been able to fix it, I found an answer in StackOverflow to install vscode-server offline, and I find that when I unpack the file with the VSCode commit id, errors occur. I mention this comment because it's similar to an error in the error log.

Source: [https://stackoverflow.com/questions/56671520/how-can-i-install-vscode-server-in-linux-offline](https://stackoverflow.com/questions/56671520/how-can-i-install-vscode-server-in-linux-offline)

Download the vscode-server-linux source code:

```
# Download url is: https://update.code.visualstudio.com/commit:${commit_id}/server-linux-x64/stable
curl -sSL "https://update.code.visualstudio.com/commit:ff915844119ce9485abfe8aa9076ec76b5300ddd/server-linux-x64/stable" -o vscode-server-linux-x64.tar.gz
```

Unzip the downloaded `vscode-server-linux-x64.tar.gz`:
```
tar -zxvf /tmp/vscode-server-linux-x64.tar.gz
```

Output from previous command:
```
vagrant@vagrant-ubuntu-trusty-64:/tmp$ tar -zxvf vscode-server-linux-x64.tar.gz
vscode-server-linux-x64/
vscode-server-linux-x64/node
tar: Skipping to next header
tar: Exiting with failure status due to previous errors
```

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: No

**Comment by roblourens** [comment]
Is it possible that you have an older version of tar? What is the output of `tar --version`?

**Comment by monoprosito** [comment]
This is the output of the `tar --version` command:

```
vagrant@vagrant-ubuntu-trusty-64:~$ tar --version
tar (GNU tar) 1.27.1
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
```

**Comment by roblourens** [comment]
I have seen this error twice before. Once was due to an old tar version, which yours seems like it should be fine, and once it went away randomly - I assume the download was corrupted.

Do you have a proxy or something that would mess with the download? What is the `md5sum`?

**Comment by monoprosito** [comment]
@roblourens Here are the answers to your questions and maybe other details that can help you get to the bottom of this.

- Do you have a proxy or something that would mess with the download?
**A:** No, the work environment where I try to run the VSCode extension is at home, the connection is direct to my ISP. There is no traffic filtering in the middle of my Vagrant and ISP.

-  What is the md5sum?
**A:** I can answer this question, but only if I download the file `vscode-server-linux-x64.tar.gz` from this URL (https://update.code.visualstudio.com/commit:ff915844119ce9485abfe8aa9076ec76b5300ddd/server-linux-x64/stable) on the folder `/tmp`.

This is the output when calculating the MD5 sum:
```
vagrant@vagrant-ubuntu-trusty-64:/tmp$ md5sum vscode-server-linux-x64.tar.gz
6b3bc185d7714e49bbf9b9d1e871adcb  vscode-server-linux-x64.tar.gz
```

I think the above MD5 sum is what you expect, because when I try to establish the SSH connection through my VSCode, I don't know in which path the _.tar.gz_ file is downloaded which is unpacked in the path `~/.vscode-server/`. Here I will leave a reference for you to better understand a context that I will mention later. **[1]**
**NOTE:** I remember reading somewhere that by the way I can't find anymore... it seems that there are two web pages (or maybe more) from which to download the file 'vscode-server-linux-x64.tar.gz'.
The only website I can remember at the moment is: https://update.code.visualstudio.com/, but I can't find the rest. I mention this because at some point you tell me that the file is possibly corrupted and maybe I assume that this could be due to the server from which I am downloading the file.

After checking your previous answer, I tried to run `sudo apt update` and `sudo apt upgrade` on my Vagrant server, in case there were new versions of the `tar` program to install, but no luck. However, I saw that Ubuntu had some updates available from **UA Infra: Extended Security Maintenance** (esm-infra) that I could apply, so I used an Ubuntu One token to apply these updates and see if anything was out of date, however after applying all these updates, nothing has changed. The program `tar` is still in version **1.27.1**.

I also wanted to test if these updates could have had any effect between the SSH connection made from the VSCode extension to my Vagrant server **[1]**, however I had no luck with that either. The error log shown to me by the VSCode console is still the same as the one I sent you at the beginning of the Issue, but I would like to show you some information that might help.

```
vagrant@vagrant-ubuntu-trusty-64:~$ ls -la
drwxrwxr-x 3 vagrant vagrant 4096 Apr 27 18:06 .vscode-server
vagrant@vagrant-ubuntu-trusty-64:~$
vagrant@vagrant-ubuntu-trusty-64:~$ ls -la .vscode-server/
drwxrwxr-x 3 vagrant vagrant 4096 Apr 27 18:05 bin
-rw-rw-r-- 1 vagrant vagrant  127 Apr 29 21:16 .ff915844119ce9485abfe8aa9076ec76b5300ddd.log
vagrant@vagrant-ubuntu-trusty-64:~$
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server$ ls -la
drwxrwxr-x 3 vagrant vagrant 4096 Apr 27 18:05 bin
-rw-rw-r-- 1 vagrant vagrant  127 Apr 29 21:16 .ff915844119ce9485abfe8aa9076ec76b5300ddd.log
vagrant@vagrant-ubuntu-trusty-64:~$
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server$ cat .ff915844119ce9485abfe8aa9076ec76b5300ddd.log
bash: line 371: /home/vagrant/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/server.sh: No such file or directory
vagrant@vagrant-ubuntu-trusty-64:~$
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server$ file .ff915844119ce9485abfe8aa9076ec76b5300ddd.log
.ff915844119ce9485abfe8aa9076ec76b5300ddd.log: ASCII text
vagrant@vagrant-ubuntu-trusty-64:~$
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server$ ls -la bin/
drwxrwxr-x 2 vagrant vagrant 4096 Apr 29 21:16 ff915844119ce9485abfe8aa9076ec76b5300ddd
vagrant@vagrant-ubuntu-trusty-64:~$
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server$ ls -la bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/
-rwxr-xr-x 1 vagrant vagrant 21789696 Apr 29 21:16 node
vagrant@vagrant-ubuntu-trusty-64:~$
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server$ file bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/node
bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/node: ERROR: ELF 64-bit LSB  executable, x86-64, version 1 (GNU/Linux), dynamically linked (uses shared libs)error reading (Invalid argument)
vagrant@vagrant-ubuntu-trusty-64:~$
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server$ ldd -v bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/node
ldd: exited with unknown exit code (135)
```

I hope this can be useful, thank you very much for your help!

**Comment by roblourens** [comment]
I got something different

```
roblou@robfast:/tmp$ ls -l vscode-server-linux-x64.tar.gz
-rw-rw-r-- 1 roblou roblou 131 May  3 09:38 vscode-server-linux-x64.tar.gz
roblou@robfast:/tmp$ md5sum vscode-server-linux-x64.tar.gz
7bd3c38c5f2a6ce443c8856f5352a425  vscode-server-linux-x64.tar.gz
```

And I assume you get the same error if you try to untar that tar that you downloaded? `tar -xf ...`

**Comment by roblourens** [comment]
I got something different

```
roblou@robfast:/tmp$ ls -l vscode-server-linux-x64.tar.gz
-rw-rw-r-- 1 roblou roblou 30003155 May  3 09:41 vscode-server-linux-x64.tar.gz
roblou@robfast:/tmp$ md5sum vscode-server-linux-x64.tar.gz
e9b2856d34f5f3a941595c3ef4e7a504  vscode-server-linux-x64.tar.gz
```

I assume that you get the same error if you try to untar the file you downloaded? `tar -xf ...`

**Comment by monoprosito** [comment]
I'm going to do that three times by downloading the same file from the same server

```
vagrant@vagrant-ubuntu-trusty-64:/tmp$ curl -sSL "https://update.code.visualstudio.com/commit:ff915844119ce9485abfe8aa9076ec76b5300ddd/server-linux-x64/stable" -o vsco de-server-linux-x64.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ curl -sSL "https://update.code.visualstudio.com/commit:ff915844119ce9485abfe8aa9076ec76b5300ddd/server-linux-x64/stable" -o vsco de-server-linux-x64_v1.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ curl -sSL "https://update.code.visualstudio.com/commit:ff915844119ce9485abfe8aa9076ec76b5300ddd/server-linux-x64/stable" -o vsco de-server-linux-x64_v2.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ ls -l
total 87900
-rw-rw-r-- 1 vagrant vagrant 30003155 May  3 21:29 vscode-server-linux-x64.tar.gz
-rw-rw-r-- 1 vagrant vagrant 30003155 May  3 21:29 vscode-server-linux-x64_v1.tar.gz
-rw-rw-r-- 1 vagrant vagrant 30003155 May  3 21:29 vscode-server-linux-x64_v2.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ md5sum vscode-server-linux-x64.tar.gz
3a9002a55ecf4a1029d4464d72211044  vscode-server-linux-x64.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ md5sum vscode-server-linux-x64_v1.tar.gz
84d9767aaa99d41075809c1062692e17  vscode-server-linux-x64_v1.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ md5sum vscode-server-linux-x64_v2.tar.gz
c030e85bd7a05e8cfe5abe7e94408aa5  vscode-server-linux-x64_v2.tar.gz
```

When trying to unpack the contents of each of these three files, this is the result:
```
vagrant@vagrant-ubuntu-trusty-64:/tmp$ tar -xf vscode-server-linux-x64.tar.gz

gzip: stdin: invalid compressed data--format violated
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
vagrant@vagrant-ubuntu-trusty-64:/tmp$ tar -xf vscode-server-linux-x64_v1.tar.gz
tar: Skipping to next header

gzip: stdin: invalid compressed data--format violated
tar: Child returned status 1
tar: Error is not recoverable: exiting now
vagrant@vagrant-ubuntu-trusty-64:/tmp$ tar -xf vscode-server-linux-x64_v2.tar.gz
tar: Skipping to next header
tar: Exiting with failure status due to previous errors
vagrant@vagrant-ubuntu-trusty-64:/tmp$ ls -l
total 87904
drwxr-xr-x 2 vagrant vagrant     4096 Apr 16 17:50 vscode-server-linux-x64
-rw-rw-r-- 1 vagrant vagrant 30003155 May  3 21:29 vscode-server-linux-x64.tar.gz
-rw-rw-r-- 1 vagrant vagrant 30003155 May  3 21:29 vscode-server-linux-x64_v1.tar.gz
-rw-rw-r-- 1 vagrant vagrant 30003155 May  3 21:29 vscode-server-linux-x64_v2.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ ls -l vscode-server-linux-x64
total 42660
-rwxr-xr-x 1 vagrant vagrant 43680144 Apr 16 17:50 node
```

You can give me the URL from where you are downloading `vscode-server-linux-x64.tar.gz`?

**Comment by monoprosito** [comment]
Effectively every time I download the `vscode-server-linux-x64.tar.gz` file with `curl` or `wget` from the https://update.code.visualstudio.com/ server I get a different MD5 sum, the problem is that this behavior only happens in **Vagrant**.

I have a temporary solution to this problem, I took as a reference an answer from StackOverflow that I will leave just below and I will detail it in the following list:

**Source:** https://stackoverflow.com/a/56781109

1. Download on the Windows host the file `vscode-server-linux-x64.tar.gz`: https://update.code.visualstudio.com/commit:ff915844119ce9485abfe8aa9076ec76b5300ddd/server-linux-x64/stable

2. Check the MD5 sum of the `vscode-server-linux-x64.tar.gz` file on the Windows host:
```
C:\Users\A Monkey Programmer\Downloads
λ ls vscode-server-linux-x64.tar.gz
vscode-server-linux-x64.tar.gz

C:\Users\A Monkey Programmer\Downloads
λ certutil -hashfile vscode-server-linux-x64.tar.gz MD5
MD5 hash of vscode-server-linux-x64.tar.gz:
e9b2856d34f5f3a941595c3ef4e7a504
CertUtil: -hashfile command completed successfully.
```
**NOTE:** There you can see that the MD5 sum corresponds to the one you showed me in your comment.

3. Send the downloaded file from the Windows host via SCP to the Vagrant machine:
```
C:\Users\A Monkey Programmer\Downloads
λ scp -i "%VAGRANTFILE_PATH%\.vagrant\machines\default\virtualbox\private_key" -P 2222 vscode-server-linux-x64.tar.gz vagrant@127.0.0.1:/tmp
vscode-server-linux-x64.tar.gz                                                                                                        100%   29MB  19.3MB/s   00:01
```

4. Move tarball to **`~/.vscode-server/bin/$COMMIT_ID/vscode-server-linux-x64.tar.gz`**
```
vagrant@vagrant-ubuntu-trusty-64:/tmp$ mv vscode-server-linux-x64.tar.gz ~/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/vscode-server-linux-x64.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ cd ~/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd/
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd$ ls -l
-rw-rw-r--  1 vagrant vagrant 30003155 May  6 18:49 vscode-server-linux-x64.tar.gz
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd$ tar -zxvf vscode-server-linux-x64.tar.gz --strip-components 1
```
After that, the file should be unpacked without problem and will show the following files in the current folder:

```
vagrant@vagrant-ubuntu-trusty-64:~/.vscode-server/bin/ff915844119ce9485abfe8aa9076ec76b5300ddd$ ls -l
total 72024
drwxr-xr-x  2 vagrant vagrant     4096 Apr 16 17:50 bin
drwxr-xr-x 30 vagrant vagrant     4096 Apr 16 17:50 extensions
-rw-r--r--  1 vagrant vagrant    13380 Apr 16 17:46 LICENSE
-rwxr-xr-x  1 vagrant vagrant 43680144 Apr 16 17:50 node
drwxr-xr-x 67 vagrant vagrant     4096 Apr 16 17:50 node_modules
drwxr-xr-x  3 vagrant vagrant     4096 Apr 16 17:50 out
-rw-r--r--  1 vagrant vagrant     1027 Apr 16 17:46 package.json
-rw-r--r--  1 vagrant vagrant    20881 Apr 16 17:46 product.json
-rwxr-xr-x  1 vagrant vagrant      222 Apr 16 17:46 server.sh
```

5. When I managed to do the previous steps, I went to Visual Studio Code and looked in **File** -> **Preferences** -> **User settings** -> **Extensions** -> **Remote SSH**.

There is an option when configuring this extension and it is called **Allow Local Server Download**, when I unchecked this option, I immediately tried to connect to the Vagrant machine and **BOOM!**. **It connected**.

**NOTE:** In **`settings.json`**, the option is **`remote.SSH.allowLocalServerDownload`** and I set the value **`false`**.

However, I am not satisfied with this solution, I know that the problem goes further, a friend who works with the same stack as me, the same thing happened to him and hopefully I could help him with this temporary solution. **At the moment I'm focused on the fact that the problem belongs to some element within Vagrant** and I'm thinking about what you mentioned about the Firewall (**ufw is disabled**), I'm also thinking that it's a problem downloading the file with **`wget`** or **`curl`**, maybe that's where the file I'm downloading is being corrupted.

Another thing to mention is that trying to install the package **`semistandard`** with **`npm`** gives me this fantastic error:

```
vagrant@vagrant-ubuntu-trusty-64:~$ npm install semistandard -g
npm ERR! code Z_DATA_ERROR
npm ERR! errno -3
npm ERR! zlib: incorrect data check
```

So, I'll keep digging!

**Comment by roblourens** [comment]
Hm, really confusing! It's weird that your 3 attempts are all the same size in bytes, but have different md5sums. Are they actually text, or can you tell anything about the data that you got?

**Comment by monoprosito** [comment]
I can show you the file type of each file:

```
vagrant@vagrant-ubuntu-trusty-64:/tmp$ ls -l
-rw-rw-r-- 1 vagrant vagrant 30003155 May  7 01:53 vscode-server-linux-x64.tar.gz
-rw-rw-r-- 1 vagrant vagrant 30003155 May  7 01:53 vscode-server-linux-x64_v1.tar.gz
-rw-rw-r-- 1 vagrant vagrant 30003155 May  7 01:53 vscode-server-linux-x64_v2.tar.gz
vagrant@vagrant-ubuntu-trusty-64:/tmp$ file vscode-server-linux-x64.tar.gz
vscode-server-linux-x64.tar.gz: gzip compressed data, from Unix, last modified: Thu Apr 16 18:02:40 2020
vagrant@vagrant-ubuntu-trusty-64:/tmp$ file vscode-server-linux-x64_v1.tar.gz
vscode-server-linux-x64_v1.tar.gz: gzip compressed data, from Unix, last modified: Thu Apr 16 18:02:40 2020
vagrant@vagrant-ubuntu-trusty-64:/tmp$ file vscode-server-linux-x64_v2.tar.gz
vscode-server-linux-x64_v2.tar.gz: gzip compressed data, from Unix, last modified: Thu Apr 16 18:02:40 2020
```

**Comment by roblourens** [comment]
Very strange. Seems like it's just getting corrupted randomly somehow then. I don't know anything about Vagrant and I don't even have a guess at what it could be.

**Comment by monoprosito** [comment]
I understand, I'll keep digging and see what I can find, it's already become a challenge for me. One last question, maybe someone I know can give me a hand?

**Comment by roblourens** [comment]
You could email me one of those files and I could see what I can figure out. But I don't know who else would have an idea, sorry.

**Comment by monoprosito** [comment]
Of course, Rob, I already sent you the email with the file you asked for!

**Comment by monoprosito** [comment]
I just received a Visual Studio Code update, I had to do the same procedure I described to you recently, only this time the commit id changed to the VSCode version.

**Comment by monoprosito** [comment]
@roblourens  I hope you're okay. I want to thank you for the time that you dedicated to this problem, I finally managed to solve it, two days ago at midnight.

The problem was with **Vagrant version 2.2.7**, I upgraded Vagrant and looked at the Release Notes which had fixed many bugs and added new features. I immediately upgraded to **version 2.2.9**, destroyed my Vagrant machine, and created a new one.

And guess what? All transparent. Everything worked as expected. I managed to install the semistandard package with npm and vscode-server without any problems.

Are you in charge of closing the issue or shall I?
